mentat/mentat_prompt_session.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+from prompt_toolkit import PromptSession
+from prompt_toolkit.application.current import get_app
+from prompt_toolkit.auto_suggest import AutoSuggestFromHistory, Suggestion
+from prompt_toolkit.buffer import Buffer
+from prompt_toolkit.completion.word_completer import WordCompleter
+from prompt_toolkit.document import Document
+from prompt_toolkit.filters import Condition
+from prompt_toolkit.history import FileHistory
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.key_binding.key_processor import KeyPressEvent
+
+from .commands import Command
+from .config_manager import mentat_dir_path
+
+
+class FilteredFileHistory(FileHistory):
+    def __init__(self, filename: str):
+        self.excluded_phrases = ["y", "n", "i", "q"]
+        super().__init__(filename)
+
+    def append_string(self, string):
+        if (
+            string.strip().lower() not in self.excluded_phrases
+            # If the user mistypes a command, we don't want it to appear later
+            and string.strip()
+            and string.strip()[0] != "/"
+        ):
+            super().append_string(string)
+
+
+class FilteredHistorySuggestions(AutoSuggestFromHistory):
+    def __init__(self):
+        super().__init__()
+
+    def get_suggestion(self, buffer: Buffer, document: Document) -> Suggestion | None:
+        # We want the auto completer to handle commands instead of the suggester
```

Comment:
what's the difference between autocompleter and suggester?

Reply:
Autocompleter shows the ghost words to the right of your cursor that you can tab to complete; the autocompleter shows like a dropdown that you can scroll through. If you just type / on this branch you'll see the autocompleter and it will make more sense

Reply:
It looks like python has a builtin library shlex.split that will automatically handle this all for us!